components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -69,8 +69,9 @@ class SystemEngineView @JvmOverloads constructor(
      * Render the content of the given session.
      */
     override fun render(session: EngineSession) {
-        this.session = session as SystemEngineSession
+        removeAllViews()
 
+        this.session = session as SystemEngineSession
         (session.webView?.parent as? SystemEngineView)?.removeView(session.webView)
         addView(initWebView(session.webView))
```

Comment:
And `@Synchronized` to protect against sessions switching fast?

Reply:
Mh. Yeah, I was just looking at this method in isolation - not thinking about how the other components may use it.